### PR TITLE
2236/script sms provider

### PR DESCRIPTION
**Pull request by cornelinux**
Add the ScriptSMSProvider, that can call arbitrary scripts, that take the phone number from the command line and the text from stdin.

### CODE REVIEW

**Comment by cornelinux** [comment]
Oups, no tests here.

File: privacyidea/lib/smsprovider/ScriptSMSProvider.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    def __init__(self, db_smsprovider_object=None, smsgateway=None, directory=None):
+        """
+        Create a new SMS Provider object fom a DB SMS provider object
+
+        :param db_smsprovider_object: The database object
+        :param smsgateway: The SMS gateway object from the database table
+            SMS gateway. The options can be accessed via
+            self.smsgateway.option_dict
+        :return: An SMS provider object
```
Comment (plettich) [inline privacyidea/lib/smsprovider/ScriptSMSProvider.py:45]: Please describe all parameters

File: privacyidea/lib/smsprovider/ScriptSMSProvider.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        """
+        self.config = db_smsprovider_object or {}
+        self.smsgateway = smsgateway
+        self.script_directory = directory or get_app_config_value("PI_SCRIPT_SMSPROVIDER_DIRECTORY",
+                                                                  "/etc/privacyidea/scripts")
+
+    def submit_message(self, phone, message):
+        """
+        send a message to a phone via an smpp protocol to smsc
```
Comment (plettich) [inline privacyidea/lib/smsprovider/ScriptSMSProvider.py:54]: We just call the script, we have no idea about the protocol, or have we?

File: privacyidea/lib/smsprovider/ScriptSMSProvider.py (hunk 1 above)
Comment (cornelinux) [reply privacyidea/lib/smsprovider/ScriptSMSProvider.py:47]: done in 8d67564e

File: privacyidea/lib/smsprovider/ScriptSMSProvider.py (hunk 2 above)
Comment (cornelinux) [reply privacyidea/lib/smsprovider/ScriptSMSProvider.py]: done in 8d67564e

**Pull request merged**